crates/flux-fhir-analysis/src/conv/mod.rs

```diff
@@ -1680,7 +1680,12 @@ impl<'genv, 'tcx: 'genv, P: ConvPhase<'genv, 'tcx>> ConvCtxt<P> {
                     )
                 }
             }
-            fhir::Res::Def(kind, def_id) => self.report_expected_type(path.span, kind, def_id)?,
+            fhir::Res::Def(kind, def_id) => {
+                match kind {
+                    DefKind::ForeignTy => flux_middle::rty::BaseTy::Foreign(def_id),
+                    _ => self.report_expected_type(path.span, kind, def_id)?,
+                }
+            }
```

Comment:
We need to check that the path has no generic arguments and raise an error if it does, similar to how we do for self-types. See, e.g., how we call `self.check_self_ty_generics`.

Also, you can inline the pattern on `DefKind::ForeignTy` as we do for the other cases.

```suggestion
            fhir::Res::Def(DefKind::ForeignTy, def_id) => rty::BaseTy::Foreign(def_id),
            fhir::Res::Def(kind, def_id) => self.report_expected_type(path.span, kind, def_id)?,
```